ONSdigital/florence
# Feature/upload component test page

### What

- Set feature flag for new generic uploads capability
- Implemented test upload page using new generic upload component developed for static files work
- Updated homepage image upload / CMD dataset uploads to make use of the new upload component. These journeys **cannot** be completed end-to-end because of additional uplift work required within the CMD import services and the image upload/import services. As these are feature flagged, there should be no impact on existing functionality in prod
- You can also test uploading via the new system, and see what resultant file metadata can be retrieved via the new Files API, by uploading a file on the [`/upload-test` route.](http://localhost:8081/florence/upload-test)

### How to review

- Check code changes
- Run publishing locally with [`dp-static-files-compose`](https://github.com/ONSdigital/dp-static-files-compose). Run Florence with `make debug ENABLE_NEW_UPLOAD=true`
- Go to homepage/CMD dataset upload and attempt to upload a file < 5 MB and > 5MB. See that both files are successfully uploaded. The upload request should be made to an `upload-new` endpoint as opposed to `upload`

### Who can review

Anyone
